File: sources/Android/android-communications/library/src/sdk/java/com/polar/sdk/api/model/PolarDeviceUuid.kt
package com.polar.sdk.api.model

object PolarDeviceUuid {
    private const val POLAR_UUID_PREFIX = "0e030000-0084-0000-0000-0000"
    private const val REQUIRED_DEVICE_ID_LENGTH = 8

    fun fromDeviceId(deviceId: String): String {
        require(deviceId.length == REQUIRED_DEVICE_ID_LENGTH) {
            "deviceId must be $REQUIRED_DEVICE_ID_LENGTH characters long, was: ${deviceId.length}"
        }
        return POLAR_UUID_PREFIX + deviceId
    }
}

File: sources/iOS/ios-communications/Sources/PolarBleSdk/sdk/api/model/PolarDeviceUuid.swift
//  Copyright © 2024 Polar. All rights reserved.

import Foundation

struct PolarDeviceUuid {
    private static let polarUuidPrefix = "0e030000-0084-0000-0000-0000"
    private static let requiredDeviceIdLength = 8
    
    enum PolarDeviceUuidError: Error {
        case invalidDeviceIdLength(expected: Int, actual: Int)
    }
    
    static func fromDeviceId(_ deviceId: String) throws -> String {
        guard deviceId.count == requiredDeviceIdLength else {
            throw PolarDeviceUuidError.invalidDeviceIdLength(expected: requiredDeviceIdLength, actual: deviceId.count)
        }
        return polarUuidPrefix + deviceId
    }
}
